src/main/java/org/broadinstitute/consent/http/service/TDRService.java

```diff
@@ -33,4 +38,18 @@ public ApprovedUsers getApprovedUsersForDataset(Dataset dataset) {
 
         return new ApprovedUsers(approvedUsers);
     }
+
+    public List<Integer> getDatasetIdsByIdentifier(List<String> identifiers) {
+        List<Integer> datasetIds = identifiers
+                .stream()
+                .filter(identifier -> !identifier.isBlank())
+                .map(identifier -> datasetDAO.findDatasetByAlias(Dataset.parseIdentifierToAlias(identifier)))
+                // technically, it is possible to have two dataset identifiers which
+                // have the same alias but are not the same: e.g., DUOS-5 and DUOS-00005
+                .filter(d -> !identifiers.contains(d.getDatasetIdentifier()))
```

Comment:
In practice, this is filtering out the dataset that was found in the map step. I'm currently looking into why.

Reply:
The conditional is backwards here, we need to know that `identifiers` DOES contain the dataset identifier.
```
.filter(d -> identifiers.contains(d.getDatasetIdentifier()))
```

Reply:
Flipped the conditional logic here to ensure we are checking that the identifiers contains the dataset identifier.